File: Core/Entities/Order/Order.cs
using System;
using System.Collections.Generic;

namespace Core.Entities.Order
{
    public class Order : BaseEntity
    {
    public Order()
    {
    }

    public Order(string shopperEmail, DeliveryAddress deliveryAddress, DeliveryMethod deliveryMethod, IReadOnlyList<OrderItem> orderItems, decimal subtotal, string orderStatus, string paymentIntentId)
    {
      ShopperEmail = shopperEmail;
      DeliveryAddress = deliveryAddress;
      DeliveryMethod = deliveryMethod;
      OrderItems = orderItems;
      Subtotal = subtotal;
      OrderStatus = orderStatus;
      PaymentIntentId = paymentIntentId;
    }

    public string ShopperEmail { get; set; }
        public DateTimeOffset OrderDate { get; set; } = DateTimeOffset.Now;
        public DeliveryAddress DeliveryAddress { get; set; }
        public DeliveryMethod DeliveryMethod { get; set; }
        public IReadOnlyList<OrderItem> OrderItems { get; set; }
        public decimal Subtotal { get; set; }
        public string OrderStatus { get; set; }
        public string PaymentIntentId { get; set; }

        public decimal GetTotal() => Subtotal + DeliveryMethod.Price;

    }
}

File: client/src/app/shared/model/order.ts
import { IBaseModel } from "./base-model";
import { IUserAddress } from "./user";

export interface IDeliveryMethod extends IBaseModel {
  deliveryTime: string;
  description: string;
  price: number;
}


export interface IOrder {
  basketId: string;
  paymentIntentId: string;
  deliveryMethodId: number;
  deliveryAddress: IUserAddress;
  orderStatus: string;
}

export interface IServerOrder extends IBaseModel{
  shopperEmail: string;
  orderDate: string;
  deliveryAddress: IUserAddress;
  deliveryMethod: string;
  deliveryPrice: number;
  orderItems: IOrderItem[];
  subtotal: number;
  total: number;
  orderStatus: string;
  paymentIntentId: string;

}

export interface IOrderItem extends IBaseModel{
  productPictureUrl: string;
  price: number;
  quantity: number;
  type: string,
  current: string,
  author: string;
}

